### PR TITLE
Provide access to tasks scheduled with the ScheduledTaskRegistrar

### PR DESCRIPTION
**Pull request by tobiasmh**
https://jira.spring.io/browse/SPR-12748

### CODE REVIEW

File: spring-context/src/main/java/org/springframework/scheduling/config/ScheduledTaskRegistrar.java [hunk 1]
```diff
@@ -122,6 +122,14 @@ public void setTriggerTasks(Map<Runnable, Trigger> triggerTasks) {
 	public void setTriggerTasksList(List<TriggerTask> triggerTasks) {
 		this.triggerTasks = triggerTasks;
 	}
+	
+	/**
+	 * Return trigger tasks as a list of {@link TriggerTask}
+	 * @since 4.2
+	 */
+	public List<TriggerTask> getTriggerTaskList() {
+		return this.triggerTasks;
```
Comment (snicoll) [inline spring-context/src/main/java/org/springframework/scheduling/config/ScheduledTaskRegistrar.java]: Regardless of whether we want to do this or not, we shouldn't expose the raw mutable list. This also means that your tests must be rewritten. 

Before doing that, let's discuss on the Jira issue what we want to do here. Thanks!

Comment (tobiasmh) [inline spring-context/src/main/java/org/springframework/scheduling/config/ScheduledTaskRegistrar.java]: If you guys are happy with moving forward on an implementation for this after the discussion takes place in Jira, I am happy to rewrite the code as needed.

Comment (snicoll) [inline spring-context/src/main/java/org/springframework/scheduling/config/ScheduledTaskRegistrar.java]: Yes, that's ok. Thanks!

**Comment by snicoll** [comment]
Merged with some updates. Thanks.

**Pull request closed without merging**